### PR TITLE
Mkulakow/chat completion request conversion

### PR DESCRIPTION
**Pull request by michalkulakowski**
### 🛠 Summary

https://jira.devtools.intel.com/browse/CVS-139240
Implementation of chat completion request conversion to HttpPayload struct.

### 🧪 Checklist

- [ ] Unit tests added.
~- [ ] The documentation updated.~ Not applicable 
- [ ] Change follows security best practices.
``

### CODE REVIEW

File: src/http_frontend/http_graph_executor_impl.cpp [hunk 1]
```diff
@@ -42,11 +42,13 @@ static const std::string UNUSED_REQUEST_ID = "";
...
     std::map<std::string, mediapipe::Packet>& inputSidePackets,  // out
-    const std::string& request) {                                // in
+    const HttpPayload& request) {                                // in
     return StatusCode::OK;
 }
 
-const std::string& getRequestId(const std::string& request) {
+const std::string& getRequestId(
+    const HttpPayload& request) {
+    static const std::string _default{""};
```
Comment (dkalinowski) [inline src/http_frontend/http_graph_executor_impl.cpp:51]: This is not needed

File: src/http_frontend/http_graph_executor_impl.cpp [hunk 2]
```diff
@@ -86,23 +88,23 @@ Status onPacketReadySerializeImpl(
...
+    std::shared_ptr<const HttpPayload> request,
     stream_types_mapping_t& inputTypes,
     PythonBackend* pythonBackend,
     ::mediapipe::CalculatorGraph& graph,
     ::mediapipe::Timestamp& currentTimestamp,
     size_t& numberOfPacketsCreated) {
     MP_RETURN_ON_FAIL(
         graph.AddPacketToInputStream(
-            "input", ::mediapipe::Adopt(new std::string(*request)).At(currentTimestamp)),
+            "input", ::mediapipe::MakePacket<HttpPayload>(*request.get()).At(currentTimestamp)),
```
Comment (mzegla) [inline src/http_frontend/http_graph_executor_impl.cpp:99]: Why are we making that change? I mean using MakePacket instead of Adopt for this one.

Comment (dkalinowski) [inline src/http_frontend/http_graph_executor_impl.cpp:99]: It does copy the HttpPayload including message string, can we avoid that?

File: src/http_rest_api_handler.cpp [hunk 3]
```diff
@@ -493,12 +494,15 @@ Status HttpRestApiHandler::processOAIChatCompletionsRequest(const HttpRequestCom
         return status;
     }
 
+    request.headers = request_components.headers;
+    request.body = request_body;
+    request.doc = &doc;
```
Comment (mzegla) [inline src/http_rest_api_handler.cpp:499]: `request.doc` is not very descriptive

File: src/http_rest_api_handler.cpp [hunk 4]
```diff
@@ -493,12 +494,15 @@ Status HttpRestApiHandler::processOAIChatCompletionsRequest(const HttpRequestCom
         return status;
     }
 
+    request.headers = request_components.headers;
+    request.body = request_body;
```
Comment (dkalinowski) [inline src/http_rest_api_handler.cpp:498]: Is it possible to use std::move so we do not copy the data?

File: src/llm/http_payload.hpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+#include <utility>
+#include <vector>
+
+#include <rapidjson/document.h>
+
+namespace ovms {
+struct HttpPayload {
+    std::vector<std::pair<std::string, std::string>> headers;
+    std::string body;          // always
+    rapidjson::Document* doc;  // pre-parsed body             = null
```
Comment (mzegla) [inline src/llm/http_payload.hpp:29]: I would go with more explicit name for this one

File: src/llm/BUILD [hunk 6]
```diff
@@ -91,3 +91,15 @@ cc_library(
     alwayslink = 1,
 )
 
+cc_library(
+    name = "httppayload",
```
Comment (mzegla) [inline src/llm/BUILD:95]: Not sure if this should be in LLM section

**Review by mzegla — APPROVED** [review]

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**